build.rs

```diff
@@ -0,0 +1,249 @@
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::{BufReader, BufWriter, Write};
+use std::path::PathBuf;
+use std::{env, io};
+
+use serde::Deserialize;
+
+enum Table {
+    Complete([char; 128]),
+    Incomplete([Option<char>; 128]),
+}
+
+struct CodeTables {
+    created: String,
+    tables: Vec<(u16, Table)>,
+}
+
+fn main() -> io::Result<()> {
+    let code_tables = parse_code_tables()?;
```

Comment:
I would add one like the below between these lines if I were you and had sufficient time:

```rust
    // Generates `code_table.rs` from `code_tables.json`
```

`///` above `fn main()` may be worse because other processes can be added in `main()` in the future.

Reply:
I extracted it into a function and added a comment. 👍